src/utils.ts

```diff
@@ -219,3 +223,54 @@ export function formatPrice(price: Price): Price {
 
   return formattedPrice;
 }
+
+
+/** 
+ * Takes a list of currentPools and newPools and returns a list
+ * of all pools that have changed in newPools
+ */
+
+export function getChangedPools(newPools: Pool[], currentPools: Pool[]) {
+  const currentPoolsMap = Object.fromEntries(currentPools.map((pool)=> {
+    return [pool.id, pool];
+  }));
+
+  return newPools.filter((pool) => {
+    if (isSame(pool, currentPoolsMap[pool.id])) {
+      return false;
+    }
+    return true;
+  });
+}
+
+export function getNonStaticSchemaFields(schema: Schema): string[] {
+  const nonStaticFields = Object.entries(schema).map(([name, details]) => {
+    if (details.static) {
+      return null;
+    }
+    return name;
+  });
+  return compact(nonStaticFields);
+}
+
+export function isSame(newPool: Pool, oldPool: Pool): boolean {
+  if (!oldPool) return false;
```

Comment:
if we check oldPool here, so provided type in arguments is wrong.
may be `oldPool?: Pool`